dbt/include/glue/macros/materializations/snapshot.sql

```diff
@@ -74,13 +74,22 @@
 
 {% macro glue__create_columns(relation, columns) %}
     {% if columns|length > 0 %}
-    {% call statement() %}
-      alter table {{ relation }} add columns (
-        {% for column in columns %}
-          `{{ column.name }}` {{ column.data_type }} {{- ',' if not loop.last -}}
-        {% endfor %}
-      );
-    {% endcall %}
+      {% set filtered_columns = [] %}
+      {% for column in columns %}
+        {% if not column.name.startswith('dbt_') %}
+          {% do filtered_columns.append(column) %}
+        {% endif %}
```

Comment:
Would you document why it needs to filter out the columns starting from `dbt_`?

Reply:
I tried to implement both unit test and integration test, however I could not make it because ...
* (unit test) test_macros.py: The target macro has internal macro statement which depends on Jinja2 template, which is difficult and tricky to mock in the unit test.
* (integ test) test_iceberg.py: I tried to add one more test case triggering "dbt snapshot" three times, and it definitely caused similar error in dbt.log but it was still very difficult to assert that specific condition inside the test case.

For now, I have reproduced the exact same issue in dbt project, and verified that the pull request successfully resolved the error.

Reply:
Thanks. I got it, approving.

Reply:
In the current implementation, when we do not filter out columns starting with `dbt_`, new meta columns called `dbt_unique_key_1`, `dbt_unique_key_2`, ...., `dbt_unique_key_N` are accidentally added to the table definition permanently, and this cause further issues in subsequent dbt snapshot runs.

Reply:
Thanks. Is it possible to write the above explanation as a comment in the source code?